### DISCUSSION TITLE
httpcheck : Is it possible to use an endpoint to get a list of endpoints to check

### DISCUSSION THREAD

**Discussion by efavry** [netdata/netdata#17046] [Q&A] [answered]
Hi y'all,
I have been wondering about advanced usage for the httpcheck module inside the go.d.plugin.
Say that I have several endpoints to test.
They never get removed and are listed by a GET /endpointsToTest
I have no way of knowing if one is added besides calling that endpoint.
Is there any way to do that regularly ? Instead of checking ourselves ?

My guess is that the answer is no and that I would either need a cronjob modifying and reloading the config file or a modified version of the module, isn't it ?

Thank you in advance for your help.

**Comment by ilyam8** [comment, marked as answer]
Hey, @efavry. You can write a script + cron that will periodically 

- do a HTTP request to ` /endpointsToTest` 
- and create/update a configuration file for the httpcheck collector.

See https://github.com/netdata/netdata/discussions/16063#discussioncomment-7123152 - it was a similar request for ping.

**Reply by efavry** [reply]
That file watching feature is great, thank you for your help !
